dipy/segment/featurespeed.pyx

```diff
@@ -222,6 +249,89 @@ cdef class CenterOfMassFeature(CythonFeature):
             out[0, d] /= N
 
 
+cdef class MidpointFeature(CythonFeature):
+    """ Extracts features from a sequential datum.
+
+    A sequence of N-dimensional points is represented as a 2D array with
+    shape (nb_points, nb_dimensions).
+
+    The feature being extracted consists in one N-dimensional point representing
+    the middle point of the sequence (i.e. `nb_points//2`th point).
+    """
+    def __init__(MidpointFeature self):
+        super(MidpointFeature, self).__init__(is_order_invariant=False)
+
+    cdef Shape c_infer_shape(MidpointFeature self, Data2D datum) nogil except *:
+        cdef Shape shape
+        shape.ndim = 2
+        shape.dims[0] = 1
+        shape.dims[1] = datum.shape[1]
+        shape.size = datum.shape[1]
+        return shape
+
+    cdef void c_extract(MidpointFeature self, Data2D datum, Data2D out) nogil:
+        cdef:
+            int N = datum.shape[0], D = datum.shape[1]
+            int mid = N/2
+            int d
+
+        for d in range(D):
+            out[0, d] = datum[mid, d]
+
+
+cdef class ArcLengthFeature(CythonFeature):
+    """ Extracts features from a sequential datum.
+
+    A sequence of N-dimensional points is represented as a 2D array with
+    shape (nb_points, nb_dimensions).
+
+    The feature being extracted consists in one scalar representing
+    the arc length of the sequence (i.e. the sum of the length of all segments).
+    """
+    def __init__(ArcLengthFeature self):
+        super(ArcLengthFeature, self).__init__(is_order_invariant=True)
+
+    cdef Shape c_infer_shape(ArcLengthFeature self, Data2D datum) nogil:
+        cdef Shape shape
+        shape.ndim = 2
+        shape.dims[0] = 1
+        shape.dims[1] = 1
+        shape.size = 1
+        return shape
+
+    cdef void c_extract(ArcLengthFeature self, Data2D datum, Data2D out) nogil:
+        out[0, 0] = c_length(datum)
+
+
+cdef class VectorBetweenEndpointsFeature(CythonFeature):
+    """ Extracts features from a sequential datum.
+
+    A sequence of N-dimensional points is represented as a 2D array with
+    shape (nb_points, nb_dimensions).
+
+    The feature being extracted consists in one N-dimensional point representing
```

Comment:
IIUC, this feature is not really a point in the same way that the inputs are sequences of points, but rather a vector in the N-dimensional space pointing from one end-point to the other.